### ISSUE TITLE
metricbeat 8.17.0 docker container not starting

### ISSUE THREAD

**Issue by wargamez** [elastic/beats#42038]
docker run docker.elastic.co/beats/metricbeat:8.17.0 
Error: unknown command "container" for "metricbeat"
Run 'metricbeat --help' for usage.

**Comment by belimawr** [comment]
I can confirm and reproduce it.

I found an easy workaround, you can add `-e` as an argument and it will run and send the logs to stderr:
```
```

**Comment by fonfalleh** [comment]
Probably the same cause [in ](https://github.com/elastic/beats/issues/42039)

**Comment by cmacknz** [comment]
Yes, the Beats are mostly all built from the same Dockerfile template.

I closed https://github.com/elastic/beats/issues/42039 as a duplicate an generalized the title of this issue so we don't end up with an issue for each unique Beat. 

Thanks for reporting this. We have many automated tests for our Docker containers, but as it turns out none that start them without adding additional argument.

The reference Kubernetes YAML includes the workaround by default. https://github.com/elastic/beats/blob/e07b5c9ed41f73446cd143b382d13f958b0037b9/deploy/kubernetes/filebeat-kubernetes.yaml#L184-L190

**Comment by wargamez** [comment]
but adding a -e would redirect all logs to stderr and not stdout according to the documentation of -e, so really it is not a fix?  A fix would be to change and fix the Dockerfile CMD commend to use --environment instead of -environment. filebeat --help says it should be with --environmen, but in the official Dockerfile and online doc -environment is used. Or fix the startup script to be able to handle -environment

Ref: https://hub.docker.com/layers/elastic/metricbeat/8.17.0/images/sha256-7ba3768050419565c3dc82ad5fa641fa7de1e5709dafbcd8aaf2c3399d589a24

docker run docker.elastic.co/beats/filebeat:8.17.0 --help
Usage:
  filebeat [flags]
  filebeat [command]

Available Commands:
  export      Export current config or index template
  generate    Generate Filebeat modules, filesets and fields.yml
  help        Help about any command
  keystore    Manage secrets keystore
  modules     Manage configured modules
  run         Run filebeat
  setup       Setup index template, dashboards and ML jobs
  test        Test config
  version     Show current version info

Flags:
  -E, --E setting=value              Configuration overwrite
  -M, --M setting=value              Module configuration overwrite
  -N, --N                            Disable actual publishing for testing
  -c, --c string                     Configuration file, relative to path.config (default "filebeat.yml")
      --cpuprofile string            Write cpu profile to file
  -d, --d string                     Enable certain debug selectors
  -e, --e                            Log to stderr and disable syslog/file output
      --environment environmentVar   set environment being ran in (default default)
  -h, --help                         help for filebeat
      --httpprof string              Start pprof http server
      --memprofile string            Write memory profile to this file
      --modules string               List of enabled modules (comma separated)
      --once                         Run filebeat only once until all harvesters reach EOF
      --path.config string           Configuration path
      --path.data string             Data path
      --path.home string             Home path
      --path.logs string             Logs path
      --plugin pluginList            Load additional plugins
      --strict.perms                 Strict permission checking on config files (default true)
  -v, --v                            Log at INFO level

Use "filebeat [command] --help" for more information about a command.

**Comment by fonfalleh** [comment]
Agreed. Using two dashes works, and differs from docs.

Potentially related commit mentioned in
https://github.com/elastic/beats/issues/42039#issuecomment-2545942247

**Comment by cmacknz** [comment]
> A fix would be to change and fix the Dockerfile CMD commend to use --environment instead of -environment.

Yes there is a draft PR for this in the works already https://github.com/elastic/beats/pull/42060/.

K8s has always used `-e` and I am over focusing on that because it is the most common use of the container, but you are right that `-e` is not the same as `--environment` and the known issue and even the docs for 8.17.0 for getting started can be updated to explicitly specifying `--environment` instead of `-e` as the work around.

**Comment by liuziyuan** [comment]
I want to know how to write -e ,when i use docker compose .yml file .

**Comment by JiffsMaverick** [comment]
@liuziyuan Add this line:
```diff
  metricbeat:
    image: docker.elastic.co/beats/metricbeat:8.17.0
    restart: unless-stopped
+   command: ["--environment", "container"]
    volumes:
```
